### PR TITLE
feat: add postWhatsappTrackingEntry api

### PR DESCRIPTION
**Pull request by talamcol**
References: [CAR-6006](https://autoricardo.atlassian.net/browse/CAR-6006)

New api to store data collected by the whatsapp modal

### CODE REVIEW

**Review by Averethel — APPROVED** [review]
Code looks good to me. If you don't feel like tackling the comment please create a follow-up tech task to refactor.

Also, please add the new API call to the README so we have track/visibility on what's implemented at a glance.

**Review by lkappeler — APPROVED** [review]
Looks good to me

**Review by ralucacril — APPROVED** [review]

**Review by masone — APPROVED** [review]

File: src/services/reporting/whatsapp.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}: {
+  listingId: number
+  whatsappEntry: WhatsappEntry
+  options?: ApiCallOptions & { validateOnly?: boolean }
+}): Promise<WithValidationError<WhatsappEntry>> => {
+  const { validateOnly, ...otherOptions } = options
+
+  const path = `listings/${listingId}/whats-app-tracking-entries${
+    validateOnly ? "/validate" : ""
+  }`
```
[the remark this answers is not in the document]
Comment (talamcol) [reply src/services/reporting/whatsapp.ts]: done

**Comment by talamcol** [comment]
@Averethel added the helper function you suggested and tested all 4 forms it affects. Is it ok to merge it?

**Comment by Averethel** [comment]
Sure, go for it

**Pull request merged**